scripts/monitor-plugin-nuts.js

```diff
@@ -0,0 +1,138 @@
+ */
+class MonitorPluginNuts {
+  pipelineData = {};
+  isComplete = false;
+  status = 'initial state';
+  justNuts = {};
+  nutsWaitInterval;
+  nutsCompletionRetryCnt;
+  constructor(pipelineData) {
+    this.pipelineData = pipelineData;
+    this.nutsCompletionRetryCnt = parseInt(process.env.NUTS_COMPLETION_RETRY_CNT ?? '30');
+    const totalWaitTime = parseInt(process.env.NUTS_WAIT_TIME ?? '900');
+    if (!this.nutsCompletionRetryCnt <= totalWaitTime) {
+      throw new Error(
+        `NUTS_WAIT_TIME ${totalWaitTime} must be greater than retry count ${this.nutsCompletionRetryCnt}`
+      );
+    }
+    this.nutsWaitInterval = Math.floor(parseInt(process.env.NUTS_WAIT_TIME ?? 900) / this.nutsCompletionRetryCnt);
+  }
+
+  getWorkflowUrl() {
+    return `${circleciBaseUrl}pipeline/${this.pipelineData.id}/workflow`;
+  }
+
+  getCircleCiUrl() {
+    return `https://app.circleci.com/pipelines/${this.justNuts.project_slug}/${this.justNuts.pipeline_number}/workflows/${this.justNuts.id}`;
+  }
+
+  async circle(url, options = {}) {
+    const defaultOptions = {
+      method: 'GET',
+      headers: { 'Circle-Token': process.env.CIRCLE_TOKEN, responseType: 'json', resolveBodyOnly: true },
+    };
+    options = Object.assign(defaultOptions, { ...options });
+    try {
+      return JSON.parse((await got(url, options)).body);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  }
+
+  async checkWorkflowState() {
+    const url = this.getWorkflowUrl();
+    const response = await this.circle(url, {});
+    const justNuts = response.items.find((item) => item.name === 'just-nuts');
+    if (!justNuts) {
+      this.isComplete = true;
+      this.status = 'success';
+      return;
+    }
+    this.isComplete = notCompleteStatus.some((status) => justNuts.status !== status);
+    this.justNuts = justNuts;
+  }
+
+  async waitForCompletion() {
+    let retryCnt = 0;
+    while (!this.isComplete && retryCnt++ <= this.nutsCompletionRetryCnt) {
+      await sleep(Duration.seconds(this.nutsWaitInterval));
+      await this.checkWorkflowState();
+    }
+    if (retryCnt > retries && !this.isComplete) {
+      this.status = 'monitor timed out';
+    }
+  }
+  displayWorkflowState() {
+    console.log(`Workflow: ${this.justNuts.name} Status: ${this.justNuts.status} URL: ${this.getCircleCiUrl()}`);
+  }
+}
+
+if (process.argv.length < 3) {
+  throw Error('did not receive expected input');
+}
+
+let args;
+try {
+  args = JSON.parse(process.argv[2]);
+} catch (error) {
+  throw new Error(`error parsing json input ${process.argv[2]}`);
+}
+
+if (!isArray(args)) {
+  throw new Error('expecting input to be a JSON array');
+}
+
+const monitors = args.map((arg) => new MonitorPluginNuts(arg));
+
+(async () => {
+  console.log(`Begin monitoring of ${monitors.length} workflows`);
+  await Promise.all(monitors.map(async (monitor) => monitor.waitForCompletion()));
```

Comment:
Nice!

Reply:
Putting on my ops hat...I'm imagining this taking a while.  And no output until they all return?  

It might be nice to know when one repo starts/completes (console.log from the waitForCompletion or the check function it calls) so you can see what we're waiting on.